mlflow/models/signature.py

```diff
@@ -340,35 +346,90 @@ def _extract_type_hints(f, input_arg_index):
     return _TypeHints(hints.get(arg_name), hints.get("return"))
 
 
-def _infer_signature_from_type_hints(func, input_arg_index, input_example=None):
-    hints = _extract_type_hints(func, input_arg_index)
-    if hints.input is None:
+def _is_context_in_predict_function_signature(*, func=None, parameters=None):
+    if parameters is None:
+        if func is None:
+            raise ValueError("Either `func` or `parameters` must be provided.")
+        parameters = inspect.signature(func).parameters
+    return (
+        # predict(self, context, model_input, ...)
```

Comment:
q:  May I ask why we have "context" as a positional argument rather than keyword argument of the predict signature? Is it just a historical reason?

Reply:
Yes it's just historical reason, we'll probably remove it soon